pkg/model/resource/resource.go

```diff
@@ -19,55 +19,136 @@ package resource
 	Plural string `json:"plural,omitempty"`
 
-	// ImportAlias is a cleaned concatenation of Group and Version.
-	ImportAlias string `json:"-"`
+	// Path is the path to the go package where the types are defined.
+	Path string `json:"path,omitempty"`
 
-	// Package is the go package of the Resource.
-	Package string `json:"package,omitempty"`
+	// API holds the information related to the resource API.
+	API *API `json:"api,omitempty"`
 
-	// Domain is the Group + "." + Domain of the Resource.
-	Domain string `json:"domain,omitempty"`
+	// Controller specifies if a controller has been scaffolded.
+	Controller bool `json:"controller,omitempty"`
+
+	// Webhooks holds the information related to the associated webhooks.
+	Webhooks *Webhooks `json:"webhooks,omitempty"`
+}
 
-	// Namespaced is true if the resource is namespaced.
-	Namespaced bool `json:"namespaced,omitempty"`
+// QualifiedGroup returns the fully qualified group name with the available information.
+func (r Resource) QualifiedGroup() string {
+	switch "" {
+	case r.Domain:
+		return r.Group
+	case r.Group:
+		return r.Domain
+	default:
+		return fmt.Sprintf("%s.%s", r.Group, r.Domain)
+	}
+}
 
-	// API holds the the api data that is scaffolded
-	API config.API `json:"api,omitempty"`
+// PackageName returns a name valid to be used por go packages.
+func (r Resource) PackageName() string {
+	if r.Group == "" {
+		return safeImport(r.Domain)
+	}
 
-	// Webhooks holds webhooks data that is scaffolded
-	Webhooks config.Webhooks `json:"webhooks,omitempty"`
+	return safeImport(r.Group)
 }
 
-// Data returns the ResourceData information to check against tracked resources in the configuration file
-func (r *Resource) Data() config.ResourceData {
-	return config.ResourceData{
-		Group:    r.Group,
-		Version:  r.Version,
-		Kind:     r.Kind,
-		API:      &r.API,
-		Webhooks: &r.Webhooks,
+// ImportAlias returns a identifier usable as an import alias for this resource.
+func (r Resource) ImportAlias() string {
+	if r.Group == "" {
+		return safeImport(r.Domain + r.Version)
 	}
+
+	return safeImport(r.Group + r.Version)
+}
+
+// HasAPI returns true if the resource has an associated API.
+func (r Resource) HasAPI() bool {
+	return r.API != nil && r.API.CRDVersion != ""
+}
+
+// HasController returns true if the resource has an associated controller.
+func (r Resource) HasController() bool {
+	return r.Controller
+}
+
+// HasDefaulting returns true if the resource has an associated defaulting webhook.
+func (r Resource) HasDefaulting() bool {
+	return r.Webhooks != nil && r.Webhooks.Defaulting
+}
+
+// HasValidation returns true if the resource has an associated validation webhook.
+func (r Resource) HasValidation() bool {
+	return r.Webhooks != nil && r.Webhooks.Validation
+}
+
+// HasConversion returns true if the resource has an associated conversion webhook.
+func (r Resource) HasConversion() bool {
+	return r.Webhooks != nil && r.Webhooks.Conversion
+}
```

Comment:
This Has should be part of the webhook struct and not resource.

Reply:
You can't check if `Resource.Webhooks` is `nil` from inside `Webhooks`, it would just error out with a nil pointer dereference error.

Reply:
The solution propsoed here is: 

- resource.HasWebhooks - returns true when the resource has webhooks
- resoirce.webhooks.HasValidation - returns true when has the validation type

> resource.HasWebhooks && resoirce.webhooks.HasValidation

OR at least

- resource.HasValidationWebhook 

See that for example `resource. HasValidation` is not good because HasValidation is part of Webhooks and not resources and because as a developer I need to check the func implementation to know that is checking if it has a webhook with the Validation type. The meaning of HasValidation is not implicity at all.

Reply:
I'm down to change them to `HasXxxxxxWebhook`